### PR TITLE
Correct spelling, code formatting, add deprecation warnings to logger methods

### PR DESCRIPTION
**Pull request by Make42**
This patch corrects spelling, a little code formatting, but mainly adds deprecation warnings to logger methods.

### CODE REVIEW

File: clearml/logger.py [hunk 1]
```diff
@@ -224,6 +228,9 @@ def report_vector(
             See full details on the supported configuration: https://plotly.com/javascript/reference/layout/
             example: extra_layout={'showlegend': False, 'plot_bgcolor': 'yellow'}
         """
+        warnings.warn(":meth:`Logger.report_vector` is deprecated;"
```
Comment (jkhenning) [inline clearml/logger.py:231]: @Make42 can you please change the layout here and in other deprecation warnings to:
```python
        warnings.warn(
            ":meth:`Logger.report_vector` is deprecated; use :meth:`Logger.report_histogram` instead.",
            DeprecationWarning,
        )
```

I'd like to stick to Black formatting 🙂

Comment (Make42) [reply clearml/logger.py]: @jkhenning Like so?

**Comment by Make42** [comment]
Since I am on it, @pollfly and @jkhenning, I corrected the `query_tasks` docstring. The description on how  `"__$not"` is to be used was erroneous - I tested it. **But**, I am not sure if it is right now. Particularly, I suspect that `"__$all"` is actually deprecated, considering https://clear.ml/docs/latest/docs/clearml_sdk/task_sdk/#tag-filters. I was not able to find anything in the code and suspect that this is implemented in the back-end. If "__$all" is deprecated, I would like to remove this part in the docstring. Also, I am only "90%" - so to speak - confident about my description. Please check!

File: clearml/automation/controller.py [hunk 2]
```diff
@@ -3623,8 +3628,9 @@ def example_retry_on_failure_callback(pipeline, node, retries):
                         # allow up to 5 retries (total of 6 runs)
                         return retries < 5
 
-        :param pre_execute_callback: Callback function, called when the step (Task) is created
-            and before it is sent for execution. Allows a user to modify the Task before launch.
+        :param pre_execute_callback: Callback function, called when the step (Task) is created,
+             and before it is sent for execution.
+             Allows a user to modify the Task before launch.
```
[the remark this answers is not in the document]
Comment (Make42) [reply clearml/automation/controller.py]: @pollfly : Better?

**Pull request merged**